web/src/typing_events.ts

```diff
@@ -104,6 +113,29 @@ export function render_notifications_for_narrow(): void {
     }
 }
 
+function apply_message_edit_notifications($row: JQuery | undefined, is_typing: boolean): void {
+    const $editing_notifications = $row?.find("#editing_notifications");
+    if (is_typing) {
+        $row?.find(".message_edit_notice").hide();
+    } else {
+        $row?.find(".typing-animation").hide();
+        $row?.find(".message_edit_notice").show();
+    }
+    $editing_notifications?.html(
+        render_editing_notifications({
+            typing: is_typing,
+        }),
+    );
+}
+
+function render_message_editing_typing(message_id: number, is_typing: boolean): void {
+    const $row = message_lists.current?.get_row(message_id);
+    apply_message_edit_notifications($row, is_typing);
+
+    const $all_msgs_view_row = message_lists.home?.get_row(message_id);
+    apply_message_edit_notifications($all_msgs_view_row, is_typing);
+}
```

Comment:
So since "All Messages" is gonna be staying all the time without re render every time you click on it, I am dealing it separately by adding/removing the typing notification each time.